src/artemis/parameters/internal_parameters.py

```diff
@@ -0,0 +1,126 @@
+        insertion_prefix: str = SIM_INSERTION_PREFIX,
+        experiment_type: str = EXPERIMENT_NAMES[0],
+        detector_params: DetectorParams = DetectorParams(**DETECTOR_PARAM_DEFAULTS),
+        ispyb_params: IspybParams = IspybParams(**ISPYB_PARAM_DEFAULTS),
+    ) -> None:
+        self.zocalo_environment = zocalo_environment
+        self.beamline = beamline
+        self.insertion_prefix = insertion_prefix
+        self.experiment_type = experiment_type
+        self.detector_params = detector_params
+        self.ispyb_params = ispyb_params
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, ArtemisParameters):
+            return NotImplemented
+        elif self.zocalo_environment != other.zocalo_environment:
+            return False
+        elif self.beamline != other.beamline:
+            return False
+        elif self.insertion_prefix != other.insertion_prefix:
+            return False
+        elif self.experiment_type != other.experiment_type:
+            return False
+        elif self.detector_params != other.detector_params:
+            return False
+        elif self.ispyb_params != other.ispyb_params:
+            return False
+        return True
+
+
+class InternalParameters:
+    artemis_params: ArtemisParameters
+    experiment_params: EXPERIMENT_TYPES
+    completeness: InternalParameterCompleteness
+
+    def __init__(self, external_params: RawParameters = RawParameters()):
+        self.artemis_params = ArtemisParameters(
+            **external_params.artemis_params.to_dict()
+        )
+        self.artemis_params.detector_params = DetectorParams(
+            **self.artemis_params.detector_params
+        )
+        self.artemis_params.detector_params.detector_size_constants = (
+            constants_from_type(
+                self.artemis_params.detector_params.detector_size_constants
+            )
+        )
+        self.artemis_params.ispyb_params = IspybParams(
+            **self.artemis_params.ispyb_params
+        )
+        self.artemis_params.ispyb_params.upper_left = Point3D(
+            *self.artemis_params.ispyb_params.upper_left
+        )
+        self.artemis_params.ispyb_params.position = Point3D(
+            *self.artemis_params.ispyb_params.position
+        )
+        self.experiment_params = EXPERIMENT_DICT[ArtemisParameters.experiment_type](
+            **external_params.experiment_params.to_dict()
+        )
+        self.completeness = self.check_completeness()
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, InternalParameters):
+            return NotImplemented
+        if self.artemis_params != other.artemis_params:
+            return False
+        if self.experiment_params != other.experiment_params:
+            return False
+        return True
+
+    def check_completeness(self) -> InternalParameterCompleteness:
+        return InternalParameterCompleteness.MINIMAL
+
+    def expand(self) -> None:
+        pass
```

Comment:
Should: These aren't used anywhere, I know they have more use in the next PR but I would argue there not used there either. Can we remove in this PR (along with `InternalParameterCompleteness`) and we can discuss the newer way of doing it in the next PR